src/Security/Security.php

```diff
@@ -1138,6 +1235,46 @@ public static function database_is_ready()
     }
 
     /**
+     * Resets the database_is_ready cache
+     */
+    public static function clear_database_is_ready()
+    {
+        self::$database_is_ready = null;
+        self::$force_database_is_ready = null;
+    }
+
+    /**
+     * For the database_is_ready call to return a certain value - used for testing
+     */
+    public static function force_database_is_ready($isReady)
+    {
+        self::$force_database_is_ready = $isReady;
+    }
+
+    /**
+     * Enable or disable recording of login attempts
+     * through the {@link LoginRecord} object.
+     *
+     * @deprecated 4.0 Use the "Security.login_recording" config setting instead
+     * @param boolean $bool
+     */
+    public static function set_login_recording($bool)
+    {
+        Deprecation::notice('4.0', 'Use the "Security.login_recording" config setting instead');
```

Comment:
Deprecation::notice('4.0') means to be removed at 4.0. Did you mean 5.0? otherwise, just remove the method directly without deprecation (my preference).

make sure you document any removed methods in 4.0.0.md changelog :)